plugins/uefi-recovery/fu-plugin-uefi-recovery.c

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2019 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include "fu-device-metadata.h"
+#include "fu-plugin-vfuncs.h"
+
+void
+fu_plugin_init (FuPlugin *plugin)
+{
+	/* make sure that UEFI plugin is ready to receive devices */
+	fu_plugin_add_rule (plugin, FU_PLUGIN_RULE_RUN_AFTER, "uefi");
+	fu_plugin_set_build_hash (plugin, FU_BUILD_HASH);
+}
+
+gboolean
+fu_plugin_startup (FuPlugin *plugin, GError **error)
+{
+	if (!fu_plugin_has_custom_flag (plugin, "requires-uefi-recovery")) {
+		g_set_error_literal (error,
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_NOT_SUPPORTED,
+				     "not required");
+		return FALSE;
+	}
+	return TRUE;
+}
+
+gboolean
+fu_plugin_coldplug (FuPlugin *plugin, GError **error)
+{
+	GPtrArray *hwids = fu_plugin_get_hwids (plugin);
+	g_autoptr(FuDevice) device = fu_device_new ();
+	fu_device_set_id (device, "uefi-recovery");
+	fu_device_set_name (device, "System Firmware ESRT Recovery");
+	fu_device_set_version (device, "1.0.0", FWUPD_VERSION_FORMAT_TRIPLET);
```

Comment:
Don't you want this firmware version to match the `bios_version` field in smbios tables for this type of device?

Reply:
Also, isn't this going to be quite confusing that the version here is triplet but the version that ESRT reads after the upgrade that adds ESRT is not triplet?

Reply:
> Don't you want this firmware version to match the bios_version field

Unfortunately not, in this specific case the version is 255.255 which is suboptimal.

> but the version that ESRT reads after the upgrade that adds ESRT is not triplet?

It will be a triplet also, no? The triplet is the default version format for UEFI devices unless there's a per-vendor or per-device quirk.

Reply:
Man this firmware sounds really broken. I really hope the next version cleans all of this up.

Reply:
One more thought - if you want absolute minimum version you should probably set it to 0.0.0

Reply:
> Man this firmware sounds really broken

I've made a giant list and cc'd all the right people. We even had `To be filled by O.E.M` which now the LVFS catches too.